### PR TITLE
[Full Screens - Registration] VerifyCodeScreen

### PR DESCRIPTION
**Pull request by manojleaton**
Fixes #BLUI-5125 .

#### Changes proposed in this Pull Request:

- Added VerifyCodeScreen.
- Updated the example

#### Screenshots / Screen Recording (if applicable)
Screenshot 2024-02-27 at 6 50 26 PM

#### To Test:

- yarn start:example-android/ios
- Click on `Registration Provider` navigation button

#### Any specific feedback you are looking for?

-

### CODE REVIEW

**Comment by ektaghag-eaton** [comment]
All props working fine. RTL and Dark theme also working fine.

**Review by ektaghag-eaton — COMMENTED** [review]
Please add test case for full screen.

File: login-workflow/src/screens/VerifyCodeScreen/VerifyCodeScreen.tsx
Comment (surajeaton) [inline login-workflow/src/screens/VerifyCodeScreen/VerifyCodeScreen.tsx:1]: When the input is valid and press Enter, it should call onNext / submit. 
You may need to update the base screen to have `onSubmitEditing` same like account details base screen.

File: login-workflow/src/screens/VerifyCodeScreen/VerifyCodeScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        onIconPress: () => navigate(-1),
+        ...WorkflowCardHeaderProps,
+    };
+
+    const workflowCardInstructionProps = {
+        instructions: t('bluiRegistration:SELF_REGISTRATION.VERIFY_EMAIL.MESSAGE'),
+        ...WorkflowCardInstructionProps,
+    };
+
+    const workflowCardActionsProps = {
```
Comment (surajeaton) [inline login-workflow/src/screens/VerifyCodeScreen/VerifyCodeScreen.tsx:132]: May need to add canGoNext as true

**Review by surajeaton — COMMENTED** [review]
Please add tests for the full screen.

**Review by surajeaton — COMMENTED** [review]
Test coverage is too low for this component.
Screenshot 2024-02-29 at 2 29 15 PM

**Review by surajeaton — COMMENTED** [review]
Please resolve the conflicts.

File: login-workflow/example/src/screens/RegistrationProviderExample.tsx [hunk 2]
```diff
@@ -22,6 +22,7 @@ import {
 // import { View } from 'react-native';
 import { RootStackParamList } from '../navigation';
 import { useNavigation } from '@react-navigation/native';
+// import { CustomScreen } from 'src/components/CustomScreen';
```
Comment (ektaghag-eaton) [inline login-workflow/example/src/screens/RegistrationProviderExample.tsx:25]: Please remove commented code if it is not going to be used in future.

File: login-workflow/src/__tests__/screens/VerifyCodeScreen-test.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+import { fireEvent, render, screen } from '@testing-library/react-native';
+import { RegistrationWorkflow } from 'src/components';
+import { RegistrationContextProvider } from 'src/contexts';
+import { VerifyCodeScreen } from 'src/screens';
+import { registrationContextProviderProps } from 'src/testUtils';
+import { SafeAreaProvider } from 'react-native-safe-area-context';
+
+jest.useFakeTimers();
+describe('Verify Code Full Screen Test cases', () => {
+    // let mockOnClose: any;
```
Comment (ektaghag-eaton) [inline login-workflow/src/__tests__/screens/VerifyCodeScreen-test.tsx:11]: Remove commented code

File: login-workflow/src/__tests__/screens/VerifyCodeScreen-test.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    let mockOnNext: any;
+
+    beforeEach(() => {
+        // mockOnClose = jest.fn();
```
Comment (ektaghag-eaton) [inline login-workflow/src/__tests__/screens/VerifyCodeScreen-test.tsx:15]: same here

File: login-workflow/src/__tests__/screens/VerifyCodeScreen-test.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                <RegistrationContextProvider {...registrationContextProviderProps}>
+                    <RegistrationWorkflow>
+                        <VerifyCodeScreen />
+                    </RegistrationWorkflow>
+                </RegistrationContextProvider>
+            </SafeAreaProvider>
+        );
+    });
+
+    // test('errorDisplayConfig onClose callBack function test', () => {
```
Comment (ektaghag-eaton) [inline login-workflow/src/__tests__/screens/VerifyCodeScreen-test.tsx:35]: Remove commented code

File: login-workflow/src/screens/index.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-export * from './DemoScreens';
+// export * from './DemoScreens';
```
Comment (ektaghag-eaton) [inline login-workflow/src/screens/index.ts:1]: Remove commented code

**Comment by ektaghag-eaton** [comment]
There is no error display on blur if the input field is empty.
Also, helper text slides down a little bit when we update the input field see attached video.

https://github.com/etn-ccis/blui-react-native-workflows/assets/133877691/a507951e-ba0d-4572-ad19-3d2f391fde70

**Comment by manojleaton** [comment]
> There is no error display on blur if the input field is empty. Also, helper text slides down a little bit when we update the input field see attached video.
> 
>  no-error-display.mp4

Once we enter code and make input field empty, then error will get display.

Screenshot 2024-03-04 at 5 38 38 PM

**Comment by manojleaton** [comment]
By mistake closed the PR, so reopened again

**Comment by ArshdeepSingh3-eaton** [comment]
everything else looks good

**Pull request merged**